projects/js-packages/connection/components/connection-status-card/index.jsx

```diff
@@ -81,6 +77,44 @@ const ConnectionStatusCard = props => {
 			} );
 	}, [ setIsFetchingConnectionData, setConnectedUserData ] );
 
+	/**
+	 * Fetch the site data on mount.
+	 * To be only run once.
+	 * Site ID is used in the disconnection flow for the optional user survey.
+	 */
+	useEffect( () => {
+		restApi
+			.fetchSiteData()
```

Comment:
When I was testing #21421 I was confused as to why I wasn't seeing the survey option when disconnecting. Looks like `fetchSiteData()` reaches out to `/wp-json/jetpack/v4/site` which is currently only registered in the Jetpack plugin. I was using just Backup & Boost plugins (Jetpack deactivated), so an `Api404Error` was being thrown here.

Reply:
Ahhh interesting. I'll take a look and see if I can find a better way to get the site ID for the survey.

Reply:
#21421 was modified to resolve this